packages/js/core/src/types/WrapError.ts

```diff
@@ -0,0 +1,334 @@
+import { CleanResolutionStep } from "../algorithms";
+
+/**
+Wrap error codes provide additional context to WrapErrors.
+
+Error code naming convention (approximate):
+    type of handler
+    type of functionality
+    piece of functionality
+    ==> handler_typeFn_pieceFn
+
+Error code map:
+  0-24 -> Client
+  25-49 -> URI resolution
+  50-74 -> Wasm wrapper invocation & sub-invocation
+  75-99 -> Plugin invocation & sub-invocation
+  100-255 -> Unallocated
+ */
+export enum WrapErrorCode {
+  UNKNOWN,
```

Comment:
Good point. I removed it.

Reply:
I'm actually "pro 0 = unknown". I do not think we should ascribe meaning to a "zero" error code, and we should treat it as "undefined". This is because often times when transmitting data, you may be reading from a buffer, and if something is not initialized properly it will be zero, when would lead to a false positive of the error we ascribe to `0`. So I think we should have `UNKNOWN` as the 0th error code.

Reply:
I added `UNKNOWN` back